src/app/GitUI/Editor/FileViewer.cs

```diff
@@ -1666,7 +1692,7 @@ public void StageSelectedLines(bool stage)
             {
                 Validates.NotNull(FilePreamble);
 
-                string treeGuid = !stage ? _viewItem.Item.TreeGuid?.ToString() : null;
+                string? treeGuid = TryUpdateTreeId(_viewItem.Item, _viewItem.SecondRevision.ObjectId)?.ToString();
                 patch = PatchManager.GetSelectedLinesAsNewPatch(
```

Comment:
The argument `treeGuid` of `GetSelectedLinesAsNewPatch` should have a more explanatory name or an xml doc. F.i. "baseTreeGuid" or "targetTreeGuid" IIUIC.

Reply:
itemBlobGuid